doc/application/application.rst

```diff
@@ -966,50 +966,60 @@ preferred, since it correctly handles dependencies between options.
 Device Tree Overlays
 ====================
 
-Zephyr uses Device Tree to describe the hardware it runs on. Sometimes it is
-necessary to modify the definitions in the :file:`.dts` files provided in the
-board folder in order to customize settings for a particular application.
-For more information on Device Tree see :ref:`device-tree`.
+As described in :ref:`device-tree`, Zephyr uses Device Tree to
+describe the hardware it runs on. This section describes how you can
+modify an application build's device tree using overlay files.
 
-The Zephyr build system takes a Device Tree configuration
```

Comment:
This text was aligned with the similar Kconfig section, now it is not, which makes it harder for the user to understand that they are similar in nature in terms of what the app can do with them?

Reply:
The build system's Kconfig and DT behaviors are different, so I don't think it's accurate to try to force them to look the same in documentation. What am I missing?

Reply:
Maybe I didn't explain myself properly. What I meant is that the Kconfig doc and the DTS doc listed the sources in reverse order of "overriding". So the first listed item's values can be overriden by later items, until the last one which cannot be overriden and hast the "last word".

Reply:
Your meaning is clear, but I believe the updated text is describing exactly that.

At cmake generation time, each device tree file is included into empty_file.c file in the order determined by the dts_files CMake variable here:

https://github.com/zephyrproject-rtos/zephyr/blob/master/cmake/dts.cmake#L34

The output of running the C pre-processor on that file is then unconditionally stored into BOARD.dts.pre.tmp. That overwrites the contents of the existing .dts.pre.tmp, rather than taking any configuration values from there, which the previous text incorrectly says it does. The macros in AUTOCONF_H are included before any device tree files in dts_files.

Scrolling up earlier in the file, dts_files contains the following, in order:

- DTS_SOURCE (which defaults to BOARD_ROOT/boards/ARCH/BOARD/BOARD.dts)
- DTS_COMMON_OVERLAYS (defaults to ZEPHYR_BASE/dts/common/common.dts)
- The contents of DTC_OVERLAY_FILE, as a list

This is the same order described in the documentation, in increasing order of ability to override. So I think this is correct and doing what you want. What am I missing?

Reply:
After reviewing this text and the `dts.cmake` file, I do believe you are correct. Thanks for the explanation.